CMakeLists.txt

```diff
@@ -137,16 +187,17 @@ target_compile_options(
                           ${BUILD_TYPE_COMPILE_FLAGS})
 target_link_options(${PROJECT_NAME} PRIVATE ${BUILD_TYPE_LINK_FLAGS})
 target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(${PROJECT_NAME} PRIVATE ${LIBRARY_NAME} ${TEST_LIBRARIES})
+target_link_libraries(${PROJECT_NAME} PRIVATE ${NESO_LIBRARY_NAME}
+                                              ${TEST_LIBRARIES})
 add_sycl_to_target(TARGET ${PROJECT_NAME} SOURCES ${MAIN_SRC})
 
 # find MPI
 find_package(MPI REQUIRED)
 # Add the NESO-Particles dependencies
 find_package(NESO-PARTICLES REQUIRED)
 
-target_link_libraries(${LIBRARY_NAME} PUBLIC ${NESO_PARTICLES_LIBRARIES})
-target_include_directories(${LIBRARY_NAME}
+target_link_libraries(${NESO_LIBRARY_NAME} PUBLIC ${NESO_PARTICLES_LIBRARIES})
```

Comment:
Probably beyond the scope of this PR, but the CMake for NESO-particles needs refactoring so that it exports its targets. That means we could just do something along the lines of `target_link_libraries(${NESO_LIBRARY_NAME} PUBLIC neso::particles)` and wouldn't need to manually set the include directories or anything.

Reply:
Created https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles/issues/35